modules/vaos/app/services/eps/provider_service.rb

```diff
@@ -89,27 +89,39 @@ def get_provider_slots(provider_id, opts = {})
     end
 
     ##
-    # Search for provider services using NPI, specialty, and address
+    # Search for provider services using NPI, specialty, and optionally address
     #
     # @param npi [String] NPI number to search for
     # @param specialty [String] Specialty to match (case-insensitive)
     # @param address [Hash] Address object with :street1, :city, :state, :zip keys
     #
     # @return OpenStruct response containing the provider service where an individual provider has
-    # matching NPI, specialty, and address, or nil if not found
+    # matching NPI and specialty, with address used as tie-breaker if multiple matches, or nil if not found
     #
     def search_provider_services(npi:, specialty:, address:)
       query_params = { npi:, isSelfSchedulable: true }
       response = perform(:get, "/#{config.base_path}/provider-services", query_params, request_headers)
 
       return nil if response.body[:provider_services].blank?
 
-      # Filter providers by specialty and address
-      matching_provider = response.body[:provider_services].find do |provider|
-        specialty_matches?(provider, specialty) && address_matches?(provider, address)
+      # Step 1: Filter providers by specialty only
+      specialty_matches = response.body[:provider_services].select do |provider|
+        specialty_matches?(provider, specialty)
       end
 
-      matching_provider&.then { |provider| OpenStruct.new(provider) }
+      return nil if specialty_matches.empty?
+
+      # Step 2: Filter specialty matches by address
```

Comment:
Note: This is done separately, so we can have some debug logs to try to identify any other formatting issues in the addresses.